p2p/gossip/protocol.go

```diff
@@ -207,7 +217,7 @@ func (p *Protocol) propagationEventLoop(ctx context.Context) {
 			}
 			metrics.PropagationQueueLen.Set(float64(len(p.propagateQ)))
 
-		case <-p.shutdown:
+		case <-p.shutdownCtx.Done():
```

Comment:
so if it was shutdown out of the select or during message processing we might never get to this path and close `pq`.

Reply:
moved into the select case. thanks